yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,191 @@
+        }
+
+
+class AcFunBangumiIE(AcFunVideoBaseIE):
+    _VALID_URL = r'(?x)https?://(?:www\.acfun\.cn/bangumi/)(?P<id>aa[_\d]+(?:\?ac=\d+)?)'
+
+    _TESTS = [{
+        'url': 'https://www.acfun.cn/bangumi/aa6002917',
+        'info_dict': {
+            'id': 'aa6002917',
+            'title': '租借女友 第1话 租借女友',
+            'duration': 1467,
+            'timestamp': 1594432800,
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': 'm3u8',
+        },
+        'skip': 'Geo-restricted to China',
+    }, {
+        'url': 'https://www.acfun.cn/bangumi/aa6002917_36188_1745457?ac=2',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        base_id, ac_idx = self._search_regex(r'(?P<id>aa[_\d]+)(?:\?ac=(?P<ac_idx>\d+))?', video_id,
+                                             'ac_idx_parse', group=['id', 'ac_idx']) or (None, None)
+        if base_id is not None:
+            video_id = f'{base_id}{format_field(ac_idx, template="_%s")}'
+
+        webpage = self._download_webpage(url, video_id)
+        json_bangumi_data = self._search_json(r'window.bangumiData\s*=\s*', webpage, 'bangumiData', video_id)
+
+        info = {}
+        if not ac_idx:
+            if 'currentVideoInfo' not in json_bangumi_data:
+                raise ExtractorError(f'Unknown webpage json schema{bug_reports_message()}')
+            video_info = json_bangumi_data['currentVideoInfo']
+            title = json_bangumi_data.get('showTitle', '')
+            season_id = json_bangumi_data.get('bangumiId')
+
+            season_number = None
+            if season_id:
+                season_number = next((
+                    idx + 1 for (idx, v) in enumerate(json_bangumi_data.get('relatedBangumis', []))
+                    if v.get('id') == season_id), 1)
+
+            json_bangumi_list = self._search_json(r'window.bangumiList\s*=\s*', webpage, 'bangumiList', video_id) or {}
+            video_internal_id = int_or_none(traverse_obj(json_bangumi_data, ('currentVideoInfo', 'id')))
+            episode_number = None
+            if video_internal_id:
+                episode_number = next(
+                    idx + 1 for (idx, v) in enumerate(json_bangumi_list.get('items', []))
+                    if v.get('videoId') == video_internal_id)
+
+            info.update({
+                'thumbnail': json_bangumi_data.get('image'),
+                'comment_count': int_or_none(json_bangumi_data.get('commentCount')),
+                'season': json_bangumi_data.get('bangumiTitle'),
+                'season_id': season_id,
+                'season_number': season_number,
+                'episode': json_bangumi_data.get('title'),
+                'episode_number': episode_number,
+            })
+        else:
+            # if has ac_idx, this url is a proxy to other video which is at https://www.acfun.cn/v/ac
+            # the normal video_id is not in json
```

Comment:
> the normal video_id is not in json

what does this mean?

Reply:
these videos should have urls like https://www.acfun.cn/v/ac000000,
but these info is not in json